openedx/core/djangolib/testing/utils.py

```diff
@@ -145,6 +146,79 @@ def setUp(self):
         self.addCleanup(self.clear_caches)
 
 
+class _AssertNumQueriesContext(CaptureQueriesContext):
+    """
+    This is a copy of Django's internal class of the same name, with the
+    addition of being able to provide a table_blacklist used to filter queries
+    before comparing the count.
+    """
+    def __init__(self, test_case, num, connection, table_blacklist=None):
+        """
+        Same as Django's _AssertNumQueriesContext __init__, with the addition of
+        the following argument:
+            table_blacklist (List): A list of table names to filter out of the
+                set of queries that get counted.
+        """
+        self.test_case = test_case
+        self.num = num
+        self.table_blacklist = table_blacklist
+        super(_AssertNumQueriesContext, self).__init__(connection)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        def is_unfiltered_query(query):
+            """
+            Returns True if the query does not contain a blacklisted table, and
+            False otherwise.
+
+            Note: This is a simple naive implementation that makes no attempt
+            to parse the query.
+            """
+            if self.table_blacklist:
+                for table in self.table_blacklist:
+                    if '"{}"'.format(table) in query['sql']:
```

Comment:
Is there any chance this could falsely grab something with the blacklisted name as a column? I'm not really familiar with the inner data workings here -- is the generated SQL the only thing we have access to?